packages/core-flows/src/product/steps/update-product-variants.ts

```diff
@@ -1,12 +1,19 @@
 import { ModuleRegistrationName } from "@medusajs/modules-sdk"
 import { IProductModuleService, ProductTypes } from "@medusajs/types"
-import { getSelectsAndRelationsFromObjectArray } from "@medusajs/utils"
+import {
+  MedusaError,
+  getSelectsAndRelationsFromObjectArray,
+} from "@medusajs/utils"
 import { StepResponse, createStep } from "@medusajs/workflows-sdk"
 
-type UpdateProductVariantsStepInput = {
-  selector: ProductTypes.FilterableProductVariantProps
-  update: ProductTypes.UpdateProductVariantDTO
-}
+type UpdateProductVariantsStepInput =
```

Comment:
thought(non-blocking): Should we separate the two operations in two steps, similar to how we have two methods in the module interface? Same comment applies to the products step.

Reply:
I am leaning towards not separating them because ultimately the step says "This step is for updating a variant", and it doesn't really matter if it is through a selector or you pass data. The fact that we use `upsert` on the module shouldn't matter for the workflow. 

I don't feel strongly about this though, so if you think it's better to separate it then that's fine with me, but I am not sure what we would name the other step 😄

Reply:
I am fine keeping the step as is, just afraid it will end up getting messy in the future. But let's address it then, instead of doing anything premature now